engine/table/src/main/java/io/deephaven/engine/table/impl/naturaljoin/typed/rightincopen/gen/RightIncrementalNaturalJoinHasherByte.java

```diff
@@ -0,0 +1,258 @@
+            final int hash = hash(k0);
+            final int firstTableLocation = hashToTableLocation(hash);
+            int tableLocation = firstTableLocation;
+            while (true) {
+                RowSet leftRowSetForState = leftRowSet.getUnsafe(tableLocation);
+                if (leftRowSetForState == null) {
+                    numEntries++;
+                    mainKeySource0.set(tableLocation, k0);
+                    final long leftRowKey = rowKeyChunk.get(chunkPosition);
+                    leftRowSet.set(tableLocation, RowSetFactory.fromKeys(leftRowKey));
+                    rightRowKey.set(tableLocation, RowSet.NULL_ROW_KEY);
+                    modifiedTrackerCookieSource.set(tableLocation, -1L);
+                    break;
+                } else if (eq(mainKeySource0.getUnsafe(tableLocation), k0)) {
+                    final long leftRowKey = rowKeyChunk.get(chunkPosition);
+                    leftRowSet.getUnsafe(tableLocation).insert(leftRowKey);
+                    break;
+                } else {
+                    tableLocation = nextTableLocation(tableLocation);
+                    Assert.neq(tableLocation, "tableLocation", firstTableLocation, "firstTableLocation");
+                }
+            }
+        }
+    }
+
+    protected void addRightSide(RowSequence rowSequence, Chunk[] sourceKeyChunks) {
+        final ByteChunk<Values> keyChunk0 = sourceKeyChunks[0].asByteChunk();
+        final LongChunk<OrderedRowKeys> rowKeyChunk = rowSequence.asRowKeyChunk();
+        final int chunkSize = keyChunk0.size();
+        for (int chunkPosition = 0; chunkPosition < chunkSize; ++chunkPosition) {
+            final byte k0 = keyChunk0.get(chunkPosition);
+            final int hash = hash(k0);
+            final int firstTableLocation = hashToTableLocation(hash);
+            boolean found = false;
+            int tableLocation = firstTableLocation;
+            RowSet leftRowSetForState;
+            while ((leftRowSetForState = leftRowSet.getUnsafe(tableLocation)) != null) {
+                if (eq(mainKeySource0.getUnsafe(tableLocation), k0)) {
+                    final long rightRowKeyForState = rightRowKey.getAndSetUnsafe(tableLocation, rowKeyChunk.get(chunkPosition));
+                    if (rightRowKeyForState != RowSet.NULL_ROW_KEY && rightRowKeyForState != QueryConstants.NULL_LONG) {
+                        final long leftRowKeyForState = leftRowSet.getUnsafe(tableLocation).firstRowKey();
+                        throw new IllegalStateException("Natural Join found duplicate right key for " + extractKeyStringFromSourceTable(leftRowKeyForState));
+                    }
+                    found = true;
+                    break;
+                }
+                tableLocation = nextTableLocation(tableLocation);
+                Assert.neq(tableLocation, "tableLocation", firstTableLocation, "firstTableLocation");
+            }
+            if (!found) {
```

Comment:
Is some action or assert expected here? (also in `removeRight()` / `addRightSide()` `modifyByRight()` / `applyRightShift()`?

Reply:
No, we don't actually care.  If there is no state (because it was never added from the left), then a missing right hand side is fine by us, we drop it to avoid holding onto extra memory.